src/Cli/dotnet/commands/dotnet-test/VSTestForwardingApp.cs

```diff
@@ -16,11 +15,24 @@ public class VSTestForwardingApp : ForwardingApp
         public VSTestForwardingApp(IEnumerable<string> argsToForward)
             : base(GetVSTestExePath(), argsToForward)
         {
+            (bool hasRootVariable, string rootVariableName, string rootValue) = GetRootVariable();
+            if (!hasRootVariable) {
+                WithEnvironmentVariable(rootVariableName, rootValue);
+            }
         }
 
         private static string GetVSTestExePath()
         {
             return Path.Combine(AppContext.BaseDirectory, VstestAppName);
         }
+
+        private static (bool hasRootVariable, string rootVariableName, string rootValue) GetRootVariable()
```

Comment:
Why don't make this internal and use it in dotnet-test as well?

Reply:
Seemed a bit weird to ask forwarding app to give us an env var. But it's probably better than having to maintain the same code in two places. Fixed.